app/models/miq_request_workflow.rb

```diff
@@ -800,14 +800,39 @@ def set_default_user_info
   def set_request_values(values)
     # Ensure that tags selected in the pre-dialog get applied to the request
     values[:vm_tags] = (values[:vm_tags].to_miq_a + @values[:pre_dialog_vm_tags]).uniq unless @values.nil? || @values[:pre_dialog_vm_tags].blank?
-
+    values[:volumes] = prepare_volumes_fields(values)
     values[:requester_group] ||= @requester.current_group.description
     email = values[:owner_email]
     if email.present? && values[:owner_group].blank?
       values[:owner_group] = User.find_by_lower_email(email, @requester).try(:miq_group_description)
     end
   end
 
+  def prepare_volumes_fields(values)
+    # the provision dialog doesn't handle arrays,
+    # so we have to hack around it to support an arbitrary
+    # number of volumes being added at once.
+    # This looks for volume form fields in the input, and converts
+    # them into an array of hashes that can be understood
+    # by prepare_volumes
+    prepare_volumes = true
+    volumes = []
+    keys = %w(volume_name volume_size volume_terminate_on_delete)
```

Comment:
There is propably a typo (swapped delete and terminate) in volume_terminate_on_delete.